src/mesh/accessors_impl.hh

```diff
@@ -274,16 +273,16 @@ inline Edge Boundary::edge()
 inline ElementAccessor<3> Boundary::element_accessor()
 {
     ASSERT_DBG(is_valid());
-    return boundary_data_->mesh_->element_accessor(boundary_data_->bc_ele_idx_);
+    return boundary_data_->mesh_->get_bc_mesh()->element_accessor(boundary_data_->bc_ele_idx_);
```

Comment:
Please, make a separate issue to remove Bounady and BoundaryData classes, replacing them by the accessors.

Reply:
Could be the next step to make Mesh simpler and more consistent.